casbin/enforcer.cpp

```diff
@@ -54,17 +54,17 @@ bool Enforcer :: enforce(string matcher, Scope scope) {
         exp_string = matcher;
 
 
-    unordered_map <string, RoleManager*> rm_map;
+    unordered_map <string, shared_ptr<RoleManager>> rm_map;
```

Comment:
the `RoleManager` won't be exposed outside `Enforcer` so use `unique_ptr` may be more appropriate

Reply:
Also I found that you are using `shared_ptr` for `Model`, `Adapter` and so on, I'm not sure which is better, do you have any good reason?